### PR TITLE
Refactor ProjectID handling

### PR DESCRIPTION
**Pull request by mrgrauel**
* Remove `ProjectId` from `Configuration` it's dynamic just like `appUserId`
* Remove `Configuration` from `Endpoints`
* Add `Configuration` to `NetworkManager`
* adds async/await method

### CODE REVIEW

File: Tests/Core/EndpointTests.swift [hunk 1]
```diff
@@ -27,59 +27,46 @@ final class EndpointTests: XCTestCase {
     }
 
     func testEnvironmentParameter() throws {
-        var configuration: Configuration = .init(appId: "1", appSecret: "2", domain: .production, projectId: "1")
-        var endpoint: Endpoint<Mock> = .init(path: "/apps/mock", method: .get(nil), configuration: configuration) { _ in
+        var endpoint: Endpoint<Mock> = .init(path: "/apps/mock", method: .get(nil)) { _ in
```
Comment (utilem) [inline Tests/Core/EndpointTests.swift]: Hier sagt Xcode:
`Variable 'endpoint' was never mutated; consider changing to 'let' constant`

File: Tests/Core/EndpointTests.swift [hunk 2]
```diff
@@ -27,59 +27,46 @@ final class EndpointTests: XCTestCase {
...
-        endpoint = .init(path: "/apps/mock", method: .get(nil), configuration: configuration) { _ in
-            Mock(name: "foobar")
-        }
-        XCTAssertEqual(endpoint.domain, .testing)
     }
 
     func testPathParameter() throws {
-        let configuration: Configuration = .init(appId: "1", appSecret: "2", domain: .production, projectId: "1")
-        var endpoint: Endpoint<Mock> = .init(path: "/apps/mock", method: .get(nil), configuration: configuration) { _ in
+        let configuration: Configuration = .init(appId: "1", appSecret: "2", domain: .production)
```
Comment (utilem) [inline Tests/Core/EndpointTests.swift]: Hier und an neuen weiteren Stellen sagt der Compiler:
`Immutable value 'configuration' was never used; consider replacing with '_' or removing it`

**Review by utilem — CHANGES_REQUESTED** [review]

File: Tests/Core/EndpointTests.swift (hunk 1 above)
Comment (mrosberghaus) [reply Tests/Core/EndpointTests.swift]: 06eb32dac989a432515f00efe7da9f21d93de975

File: Tests/Core/EndpointTests.swift (hunk 2 above)
Comment (mrosberghaus) [reply Tests/Core/EndpointTests.swift]: 06eb32dac989a432515f00efe7da9f21d93de975

**Review by utilem — APPROVED** [review]

**Pull request merged**